networkmanager: use modify_system=true

worldofpeace opened the pull request:
This does the following:
Allow users to modify system connections [0].

Technically what it does is change the NM_MODIFY_SYSTEM_POLICY
to "yes" instead of "auth_admin_keep" which would require a password prompt.
That is inacted in the following polkit policy [1].

This allows us to drop a permissive looking polkit rule, and we don't
have to instruct users to add themselves to the "networkmanager" group.

All relevant documentation is updated.

[0]: https://gitlab.freedesktop.org/NetworkManager/NetworkManager/blob/1.20.4/meson_options.txt#L17
[1]: https://gitlab.freedesktop.org/NetworkManager/NetworkManager/blob/1.20.4/data/org.freedesktop.NetworkManager.policy.in.in#L115

<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#sec-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

worldofpeace committed:
networkmanager: use modify_system=true

This does the following:
Allow users to modify system connections [0].

Techinically what it does is change the NM_MODIFY_SYSTEM_POLICY
to "yes" instead of "auth_admin_keep" which would require a password prompt.
That is inacted in the following polkit policy [1].

This allows us to drop a permissive looking polkit rule, and we don't
have to instruct users to add themselves to the "networkmanager" group.

All relevant documentation is updated.

[0]: https://gitlab.freedesktop.org/NetworkManager/NetworkManager/blob/1.20.4/meson_options.txt#L17
[1]: https://gitlab.freedesktop.org/NetworkManager/NetworkManager/blob/1.20.4/data/org.freedesktop.NetworkManager.policy.in.in#L115

jtojnar commented:
I am not sure this is something we want. Won't then every user including the system ones be able to modify network configuration?

worldofpeace commented:
It appears the rule applied to modemanager as well
```
 polkit.addRule(function(action, subject) {
      if (
        subject.isInGroup("networkmanager")
        && (action.id.indexOf("org.freedesktop.NetworkManager.") == 0
            || action.id.indexOf("org.freedesktop.ModemManager")  == 0
        ))
          { return polkit.Result.YES; }
    });
```

so I need to see if there's any build config for that as well.

worldofpeace committed:
modemmanager: use polkit permissive

This makes MM_DEFAULT_USER_POLICY = "yes" [0] [1]

[0]: https://gitlab.freedesktop.org/mobile-broadband/ModemManager/blob/8a652179deb5d0e496dd084d0a18ac457c833f62/data/org.freedesktop.ModemManager1.policy.in.in#L26
[1]: https://gitlab.freedesktop.org/mobile-broadband/ModemManager/blob/8a652179deb5d0e496dd084d0a18ac457c833f62/configure.ac#L298

worldofpeace commented:
Ok, I believe this is ready now. Probably needs a tester.
(no one volunteers it'll probably be me, though that does make the SPOF me :rofl: )

jtojnar commented on pkgs/tools/networking/modem-manager/default.nix:
```diff
@@ -24,6 +24,7 @@ stdenv.mkDerivation rec {
     "--localstatedir=/var"
     "--with-systemd-suspend-resume"
     "--with-systemd-journal"
+    "--with-polkit=permissive"
```
I am still not sure this is best practice. Though Fedora disables it completely: https://src.fedoraproject.org/rpms/ModemManager/blob/64f08c48384af5be55b8f9d668ca03613ced3780/f/ModemManager.spec#_106

worldofpeace replied:
I thought permissive made the most sense with what this PR does.
Other distros that did this included Arch Linux and Void Linux.

worldofpeace commented:
Yeah @jtojnar, I'm not sure if there's a best practice here.
Meaning, giving permissive functionality to users of a particular group `networkmanager` or just permit system users.
(and I can't remember from my polkit research if that was just wheel users)

worldofpeace commented:
Not stale, just needs a decision.
